xarray/backends/api.py

```diff
@@ -265,6 +266,11 @@ def _protect_dataset_variables_inplace(dataset, cache):
             variable.data = data
 
 
+def _protect_datatree_variables_inplace(tree, cache):
```

Comment:
```suggestion
def _protect_datatree_variables_inplace(tree: DataTree, cache: bool) -> None:
```

Reply:
we might want to type `_protect_dataset_variables_inplace` in a similar way, then